dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1535,17 +1553,48 @@ class CompileTest {
     };
   }
 
-  static Future<Map<String, dynamic>> getSizesFromIosApp(String appPath) async {
-    // Thin the binary to only contain one architecture.
-    final String xcodeBackend = path.join(flutterDirectory.path, 'packages', 'flutter_tools', 'bin', 'xcode_backend.sh');
-    await exec(xcodeBackend, <String>['thin'], environment: <String, String>{
-      'ARCHS': 'arm64',
-      'WRAPPER_NAME': path.basename(appPath),
-      'TARGET_BUILD_DIR': path.dirname(appPath),
-    });
+  static Future<Map<String, dynamic>> getSizesFromDarwinApp({
+    required String appPath,
+    required DeviceOperatingSystem operatingSystem,
+  }) async {
+    if(!operatingSystem.isDarwin) {
+      throw Exception('$operatingSystem is not a darwin os.');
+    }
+
+    late final String frameworkDirectory;
+    if (operatingSystem == DeviceOperatingSystem.ios) {
+      frameworkDirectory = path.join(
+        appPath,
+        'Frameworks',
+      );
+    } else if (operatingSystem == DeviceOperatingSystem.macos) {
+      frameworkDirectory = path.join(
+        appPath,
+        'Contents',
+        'Frameworks',
+      );
+    }
 
-    final File appFramework = File(path.join(appPath, 'Frameworks', 'App.framework', 'App'));
-    final File flutterFramework = File(path.join(appPath, 'Frameworks', 'Flutter.framework', 'Flutter'));
+    final File appFramework = File(path.join(
+      frameworkDirectory,
+      'App.framework',
+      'App',
+    ));
+
+    late final File flutterFramework;
+    if (operatingSystem == DeviceOperatingSystem.ios) {
+      flutterFramework = File(path.join(
+        frameworkDirectory,
+        'Flutter.framework',
+        'Flutter'
+      ));
+    } else if (operatingSystem == DeviceOperatingSystem.macos) {
+      flutterFramework = File(path.join(
+        frameworkDirectory,
+        'FlutterMacOS.framework',
+        'FlutterMacOS'
+      )); // https://github.com/flutter/flutter/issues/70413 😞
```

Comment:
Ha, the 😞 was for your benefit, probably shouldn't leave it as a comment since it causes more questions than answers.